stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/FileAttachmentsView.kt

```diff
@@ -59,16 +59,23 @@ internal class FileAttachmentsView : RecyclerView {
 
     private fun init(attrs: AttributeSet?) {
         style = FileAttachmentViewStyle(context, attrs)
+    }
+
+    fun setAttachments(attachments: List<Attachment>) {
         fileAttachmentsAdapter = FileAttachmentsAdapter(
```

Comment:
🤔 We are going to recreate adapter every time we set the attachments. Can we avoid doing that?
Looks like nullable listeners are handled in both `FileAttachmentsAdapter` and `FileAttachmenViewHolder` - maybe we can do something like that in `init` method:
```kotlin
        fileAttachmentsAdapter = FileAttachmentsAdapter(
            attachmentClickListener = attachmentClickListener,
            attachmentLongClickListener = attachmentLongClickListener,
            attachmentDownloadClickListener = attachmentDownloadClickListener,
            style,
        )
        adapter = fileAttachmentsAdapter
```

Reply:
During `init` the click listeners are all null, so the real behaviour won't be passed along =|. Before, the click listener was created and just made a reference to the actual listeners, making it lazy. In this case, I would like to control that the no click listener is passed along if it wasn't set. I make a change so it is only set once, please take a look.